text/0000-build-plan-merging.md

```diff
@@ -0,0 +1,260 @@
+  source = "<other dependency constraint source>"
+[requires.capabilities]
+  # arbitary keys to bools
+  build = false
+  launch = false
+[requires.modifiers]
+  # arbitrary keys and values
+  some-other-modifer = "other modifier"
+```
+
+This proposal adds a new field to `provides`:
+- `provides.strategy.version` field holds a `collect` key which indicates if the version fields from all of the requires should be merged together or if they should collected into an array.
+  - This field is structured to be extensible to other fields in `requires`.
+
+This proposal also removes both the `requires.version` and `requires.metadata` fields and replaces them with three new fields each with their own purpose:
+- `requires.version` field holds all of the version requirement information.
+  - `constraint` can be any semver constraint used to decide which dependency the providing buildpack will install.
+  - `source` is the origin of the constraint (i.e. it came from `package.json`).
+- `requires.capabilities` holds the `build` and `launch` keys and indicate when during which phase buildpack specific data and dependencies are available.
+- `requires.modifiers` is meant to hold only specific and **unique** buildpack communications.
+
+
+## Buildpack Plan (TOML) changes
+For the same reasons, we would like to add the `version`, `capabilities`, and `modifiers` fields to the Buildpack Plan, along with a `versions` field and some additional format changes:
+
+```toml
+[[entries]]
+name = "<dependency name>"
+[entries.version]
+  constraint = "<dependency constraint>"
+[entries.capabilities]
+  # or merged bool values from arbitray keys
+  build = false
+  launch = false
+[entries.modifiers]
+  # last-in-wins merged values from arbitrary keys
+  some-modifer = "modifier"
+[[entries.collection]]
+[entries.collection.version]
+  constraint = "<dependency constraint>"
+  source = "<dependency constraint source>"
+```
+In each entry the `name` field is **unique**.
+
+We arrived at this format for several reasons:
+- Moves some common buildpack operations into the lifecycle.
+  - Collects all  of the data from various `requires` entries into a unique entry for a given dependency.
+  - Allows for the implementation of information merging strategies for the `version`, `capabilities`, and `modifiers`.
+    - All version constraints are merged together into the `version` field.
+      - If the merge would result in a valid semver constraint, then image build will continue.
+      - If the merge fails to produce a valid semver constraint (i.e. a constraint the is unreachable such as "1.10.\*, 1.12.\*"), then it would cause the image build to fail.
+    - All `capabilities.build` and `capabilities.launch` for a given dependency will be `or` merged together.
+    - All keys in `modifiers` will be merged with a last in wins strategy.
+- The `collection` list will only contain elements if the `provides` strategy was set to a `collect = true`.
```

Comment:
Is the default value `false` or is this a required field?

In addition, what's the use case for when this is `false`?